### DISCUSSION TITLE
Vercel preview deployment domain error with Firebase auth

### DISCUSSION THREAD

**Discussion by pulkitvyas08** [firebase/firebase-js-sdk#6359] [Q&A]
I use Firebase for auth of my NextJS application that is deployed on Vercel. Now for every preview link that is generated by pull requests, I need to add the new domain to the auth section of the Firebase console otherwise this is the error I get:

```
Info: The current domain is not authorized for OAuth operations. This will prevent signInWithPopup, signInWithRedirect, linkWithPopup and linkWithRedirect from working. Add your domain (horse-s-mouth-website.vercel.app) to the OAuth redirect domains list in the Firebase console -> Auth section -> Sign in method tab.
```

And auth actions cannot be performed. Is there any way to automatically add the preview domains from Vercel to the Firebase auth domain list?

Now I know that we can add custom domains to the authorized domains list on firebase but how do you automate that for something like Vercel preview links where you will have a new preview link for every PR or re-deployment?

**Comment by ggukggukee** [comment]
Another workaround is create new project with github repository and change branch from main to dev in settings. 
After another git push your app will be deployed from dev. It will solve your problem.
